### PR TITLE
[INFRA-8588] Medic stability improvements

### PR DESCRIPTION
**Pull request by dblotsky**
Working around cordova entrypoint on iOS. Making medic just run 'cordova build' and 'cordova run' for iOS.

### CODE REVIEW

File: src/build/makers/ios.js [hunk 1]
```diff
@@ -15,63 +15,44 @@ module.exports = function(output,lib_location,test_dir, sha, devices, entry_poin
...
         console.log('[IOS] ' + msg + ' (sha: ' + sha + ')');
     }
-    if (keychain_location.length === 0 || keychain_password.length === 0) {
-        log('No keychain information. Fill that out in config.json if you want to build for iOS.');
+    log('starting iOS prepare');
+    try {
+        var projectWww = path.join(output, 'www');
+        // add the medic configuration (sha,host) to destination folder
+        var medic_config='{"sha":"'+sha+'","couchdb":"'+couchdb_cfg.host+'","couchdbext":"'+couchdb_cfg.exthost+'"}';
+        fs.writeFileSync(path.join(output, 'www','autotest','pages', 'medic.json'),medic_config,'utf-8');
```
Comment (vladimir-kotikov) [inline src/build/makers/ios.js]: I thought that `plugin-test-framework` searches for medic.json at `www` folder (https://github.com/apache/cordova-medic/blob/master/src/build/makers/windows.js#L84-L86)

File: src/build/makers/ios.js [hunk 2]
```diff
@@ -15,63 +15,44 @@ module.exports = function(output,lib_location,test_dir, sha, devices, entry_poin
...
+    // compile
+    log('Compiling.');
+    log(output);
 
-            //  modify config.xml
-            var configFile = path.join(output, 'mobilespec', 'config.xml');
-            fs.writeFileSync(configFile, fs.readFileSync(configFile, 'utf-8').replace(/<content\s*src=".*"/gi, '<content src="'+entry_point+'"'), 'utf-8');
-            // make sure the couch db server is whitelisted
-            fs.writeFileSync(configFile, fs.readFileSync(configFile, 'utf-8').replace(/<access origin="http:..audio.ibeat.org" *.>/gi,'<access origin="http://audio.ibeat.org" /><access origin="'+couchdb_cfg.host+'" />', 'utf-8'));
+    var mobileSpecDir = path.join(output, '../', '../');
```
Comment (vladimir-kotikov) [inline src/build/makers/ios.js]: Just nitpick: there is no need to add slashes to paths being joined. `path.join` will handle separators by itself.

File: src/build/makers/ios.js (hunk 1 above)
Comment (purplecabbage) [inline src/build/makers/ios.js]: Yes, medic.json should live in the www folder.

Comment (dblotsky) [inline src/build/makers/ios.js]: Indeed. Where does `medic.json` exist? I've never seen that file.

**Comment by dblotsky** [comment]
Comments addressed in latest commit.

File: src/build/makers/ios.js (hunk 1 above)
Comment (purplecabbage) [inline src/build/makers/ios.js]: It is a configuration file that tells the test runner where to post to completed results.  It lives in the www folder.

Comment (vladimir-kotikov) [inline src/build/makers/ios.js]: And it doesn't exist in any repo and being created before running the tests, like here (https://github.com/apache/cordova-medic/blob/master/src/build/makers/windows.js#L86)

Comment (dblotsky) [inline src/build/makers/ios.js]: So, I ran this on iOS, and that file actually never comes into existence. It seems like the CouchDB URI gets passed elsewhere.

**Pull request merged**